src/runner.py

```diff
@@ -15,6 +15,19 @@
 check_exercise_lock = Lock()
 
 
+async def single_exercise_check(exercise_path: Path):
+    if check_exercise_lock.locked():
+        return
+    with check_exercise_lock:
```

Comment:
You are right it is only used in watch mode, I'm changing this

Reply:
Well it's not that easy since we mix a coroutine and a thread lock, could not find an elegant way to remove it. It won't be a big issue to leave it even in verify mode imho.